diff --git a/node_modules/@signalwire/docusaurus-plugin-llms-txt/lib/processing/route-processor.js b/node_modules/@signalwire/docusaurus-plugin-llms-txt/lib/processing/route-processor.js
index 9c92f64..11d3f6f 100644
--- a/node_modules/@signalwire/docusaurus-plugin-llms-txt/lib/processing/route-processor.js
+++ b/node_modules/@signalwire/docusaurus-plugin-llms-txt/lib/processing/route-processor.js
@@ -27,7 +27,7 @@ async function processSingleRoute(route, cachedRoute, config, directories, logge
         return {};
     }
     try {
-        const fullHtmlPath = path.join(directories.docsDir, cachedRoute.htmlPath);
+        const fullHtmlPath = path.join(directories.docsDir, cachedRoute.htmlPath.replace("react-native-keyboard-controller/", "").replace("/index.html", ".html"));
         const effectiveConfig = getEffectiveConfigForRoute(route.path, config);
         const doc = await processHtmlFileWithContext(fullHtmlPath, route.path, effectiveConfig, directories.mdOutDir, logger, siteUrl, outDir);
         if (doc) {
