diff --git a/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp b/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
index 0000000..0000000 100644
--- a/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
+++ b/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
@@ -289,7 +289,7 @@
 bool ReactNativeFeatureFlags::preventShadowTreeCommitExhaustion() {
-  return getAccessor().preventShadowTreeCommitExhaustion();
+  return true;
 }
 
 bool ReactNativeFeatureFlags::redBoxV2Android() {
   return getAccessor().redBoxV2Android();
 }
