diff --git a/node_modules/react-native/.DS_Store b/node_modules/react-native/.DS_Store
new file mode 100644
index 0000000..597365c
Binary files /dev/null and b/node_modules/react-native/.DS_Store differ
diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm
index e74500f..c2d4515 100644
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm
@@ -68,6 +68,8 @@ @implementation RCTTextInputComponentView {
    * later comparison insensitive to them.
    */
   NSDictionary<NSAttributedStringKey, id> *_originalTypingAttributes;
+  
+  BOOL _hasInputAccessoryView;
 }
 
 #pragma mark - UIView overrides
@@ -590,10 +592,12 @@ - (void)setDefaultInputAccessoryView
        keyboardType == UIKeyboardTypeDecimalPad || keyboardType == UIKeyboardTypeASCIICapableNumberPad) &&
       containsKeyType;
 
-  if ((_backedTextInputView.inputAccessoryView != nil) == shouldHaveInputAccessoryView) {
+  if (_hasInputAccessoryView == shouldHaveInputAccessoryView) {
     return;
   }
   
+  _hasInputAccessoryView = shouldHaveInputAccessoryView;
+
   if (shouldHaveInputAccessoryView) {
     NSString *buttonLabel = [self returnKeyTypeToString:returnKeyType];
 
