diff --git a/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp b/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
index 667aec1..5cd9376 100644
--- a/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
+++ b/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
@@ -215,7 +215,7 @@ double ReactNativeFeatureFlags::preparedTextCacheSize() {
 }
 
 bool ReactNativeFeatureFlags::preventShadowTreeCommitExhaustion() {
-  return getAccessor().preventShadowTreeCommitExhaustion();
+  return true;
 }
 
 bool ReactNativeFeatureFlags::traceTurboModulePromiseRejectionsOnAndroid() {
