diff --git a/node_modules/react-native-gesture-handler/RNGestureHandler.podspec b/node_modules/react-native-gesture-handler/RNGestureHandler.podspec
index 100d320..faf33af 100644
--- a/node_modules/react-native-gesture-handler/RNGestureHandler.podspec
+++ b/node_modules/react-native-gesture-handler/RNGestureHandler.podspec
@@ -19,7 +19,7 @@ Pod::Spec.new do |s|
   if fabric_enabled
     # folly_version must match the version used in React Native
     # See folly_version in react-native/React/FBReactNativeSpec/FBReactNativeSpec.podspec
-    folly_version = '2021.06.28.00-v2'
+    folly_version = '2021.07.22.00'
     folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
 
     s.pod_target_xcconfig = {
