diff --git a/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp b/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
index 8e65a96..bf113f7 100644
--- a/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
+++ b/node_modules/react-native/ReactCommon/react/featureflags/ReactNativeFeatureFlags.cpp
@@ -287,7 +287,7 @@ double ReactNativeFeatureFlags::preparedTextCacheSize() {
 }
 
 bool ReactNativeFeatureFlags::preventShadowTreeCommitExhaustion() {
-  return getAccessor().preventShadowTreeCommitExhaustion();
+  return true;
 }
 
 bool ReactNativeFeatureFlags::redBoxV2Android() {
